Added alternative palette with colors from Monokai Pro

kmaasrud opened the pull request:
I quite like the colors in Monokai Pro as well, so I added them as an optional palette. I also added a description on how to apply it in the README.

The changes are sadly not backwards compatible, users will have to run `require('monokai').setup()` instead of just `require('monokai')`. The changes are documented in the README.

Feel free to make any changes if you disagree with my implementation, or discard the PR alltogether. I added these colors mainly for myself, but I thought I should share my work nonetheless :smile:

tanvirtin commented on colors/monokai.vim:
```diff
@@ -1 +0,0 @@
-lua require('monokai')
```
We should change this to `lua require('monokai').setup()`, this will allow the plugins to be used regardless of whether or not users call the `setup` method.

Users who want to use a different flavour of monokai can call the following in their dotfiles:
```
local monokai = require('monokai')
monokai.setup(monokai.palette_pro)
```

kmaasrud replied:
Fixed in b401f9c

tanvirtin commented on README.md:
```diff
@@ -1,25 +1,48 @@
 syntax on
-colorscheme monokai
 set termguicolors
+lua require('monokai').setup()
```
If we add the line suggested in monokai.vim, we can remove this line from the README.md :)

tanvirtin replied:
<img width="1792" alt="monokai_pro" src="https://user-images.githubusercontent.com/25164326/121989496-3f782d00-cd6a-11eb-9d60-a9fd5e16a87b.png">

We can also add the screenshot above as a sneak preview for this new feature.

kmaasrud replied:
New method documented in README, see 4c203fc

tanvirtin reviewed (changes requested):
@kmaasrud This is great!! The colorscheme looks beautiful. There are a lot of different flavours of monokai out there. It's an excellent idea for this plugin  to be able to facilitate different renditions.

I just left a few changes, to hopefully make this new feature backward compatible, let me know if these ideas work, thanks!

kmaasrud commented:
The requested changes are now implemented. The familiar `colorscheme <name>` interface is probably a better solution for those using `init.vim`, I agree. The palettes are now separate color schemes, which means one can change to Monokai Pro simply by running `colorscheme monokai_pro`.

I also noticed that the colors I initially used were a bit wrong, so they are now corrected to a palette fetched from the official site.
